call-automation-appointment-reminder/program.ts

```diff
@@ -0,0 +1,350 @@
+            "Recognition timed out for call connection id: " +
+              eventType.callConnectionId
+          );
+          playSource.uri =
+            callConfiguration.appBaseUri + callConfiguration.TimedoutAudio;
+
+          //Play audio for time out
+          await callConnectionMedia.playToAll(playSource, {
+            operationContext: "ResponseToDtmf",
+            loop: false,
+          });
+        }
+      }
+      if (eventType.kind == "PlayCompleted") {
+        Logger.logMessage(
+          MessageType.INFORMATION,
+          "PlayCompleted event received for call connection id: " +
+            eventType.callConnectionId
+        );
+        await callConnection.hangUp(true);
+      }
+      if (eventType.kind == "PlayFailed") {
+        Logger.logMessage(
+          MessageType.INFORMATION,
+          "PlayFailed event received for call connection id: " +
+            eventType.callConnectionId
+        );
+        await callConnection.hangUp(true);
+      }
+    } else {
+      return;
+    }
+  });
+  // return Results.Ok();
+}
+/// <summary>
+/// Fetch configurations from App Settings and create source identity
+/// </summary>
+/// <param name="appBaseUrl">The base url of the app.</param>
+/// <returns>The <c CallConfiguration object.</returns>
+function initiateConfiguration(appBaseUrl: string) {
+  var connectionString = configuration.ConnectionString;
+  var sourcePhoneNumber = configuration.SourcePhoneNumber;
+  var targetPhoneNumber = configuration.TargetPhoneNumber;
+  var appBaseUri = appBaseUrl;
+  var eventCallBackRoute = configuration.EventCallBackRoute;
+  var appointmentReminderMenuAudio = configuration.AppointmentReminderMenuAudio;
+  var appointmentConfirmedAudio = configuration.AppointmentConfirmedAudio;
+  var appointmentCancelledAudio = configuration.AppointmentCancelledAudio;
+  var invalidInputAudio = configuration.InvalidInputAudio;
+  var timedoutAudio = configuration.TimedoutAudio;
+  return new CallConfiguration(
+    connectionString,
+    sourcePhoneNumber,
+    targetPhoneNumber,
+    appBaseUri,
+    eventCallBackRoute,
+    appointmentReminderMenuAudio,
+    appointmentConfirmedAudio,
+    appointmentCancelledAudio,
+    invalidInputAudio,
+    timedoutAudio
+  );
+}
+function GetAudioForTone(
+  toneDetected: DtmfTone,
+  callConfigurationForTone: CallConfiguration
+) {
+  let playSource: FileSource = { uri: "" };
+
+  if (toneDetected == DtmfTone.One) {
+    playSource.uri =
+      callConfigurationForTone.appBaseUri +
+      callConfigurationForTone.appointmentConfirmedAudio;
+  } else if (toneDetected == DtmfTone.Two) {
+    playSource.uri =
+      callConfigurationForTone.appBaseUri +
+      callConfigurationForTone.appointmentCancelledAudio;
+  } // Invalid Dtmf tone
+  else {
+    playSource.uri =
+      callConfigurationForTone.appBaseUri +
+      callConfigurationForTone.invalidInputAudio;
+  }
+
+  return playSource;
+}
+function getIdentifierKind(participantnumber: string) {
```

Comment:
instead of this, you can use isCommunicationUserIdentifier and isPhoneNumberIdentifier methods